web/src/main/java/com/marklogic/hub/web/service/MappingManagerService.java

```diff
@@ -99,4 +100,8 @@ public MappingModel getMapping(String mappingName, boolean createIfNotExisted) t
         }
     }
 
+    public JsonNode getMappingFunctions() {
```

Comment:
I don't see the point in these *Service objects. It seems simpler with less indirection for MappingController to autowire in the instance of HubConfig and then create MappingFunctions itself. It's one line of code then in the controller:

    return new MappingFunctions(hubConfig.newStagingClient()).getMappingFunctions();

Reply:
I didn't want to break the precedence that we had . I will make that change

Reply:
I think next time we need code like this, it'd be worth cleaning up the other areas that use a results iterator instead of just using JacksonHandle.